numba/targets/arraymath.py

```diff
@@ -1365,9 +1364,9 @@ def _prepare_cov_input_inner(m, y, rowvar, dtype):
             y_rows, y_cols = y_arr.shape
 
             if m_cols != y_cols:
-                raise ValueError('m and y must have the same number of variables')
-                # 'variables' as the constraint on rows or columns depends on
-                # whether rowvar is True or False...
+                msg = ("m and y have incompatible dimensions and thus "
+                       "cannot be concatenated")
```

Comment:
Can probably drop the part about concatenation, this is an implementation detail?

Reply:
Done